pkg/logbridge/logbridge_test.go

```diff
@@ -0,0 +1,50 @@
+package logbridge
+
+import (
+	"bufio"
+	"bytes"
+	"math/rand"
+	"testing"
+	"time"
+)
+
+func TestLogBridge(t *testing.T) {
+	type testCase struct {
+		inputSize      int
+		bridgeCapacity int
+		expected       int
+	}
+
+	testCases := []testCase{
+		testCase{inputSize: 0, bridgeCapacity: 0, expected: 0},
+		testCase{inputSize: 10, bridgeCapacity: 10, expected: 10},
+		testCase{inputSize: 100, bridgeCapacity: 10, expected: 10},
+		testCase{inputSize: 10, bridgeCapacity: 100, expected: 10},
+	}
+
+	for i, testCase := range testCases {
+		r := rand.New(rand.NewSource(time.Now().Unix()))
+		input := make([]byte, 0, 2*testCase.inputSize)
+		for i := 0; i < testCase.inputSize; i++ {
+			input = append(input, byte(r.Intn(256)), byte(10))
+		}
+
+		reader := bytes.NewReader(input)
+		buf := new(bytes.Buffer)
+		writer := bufio.NewWriter(buf)
+
+		bridge := New(reader, writer, testCase.bridgeCapacity)
+
+		if err := bridge.Run(); err != nil {
+			t.Fatal(err)
+		}
+
+		if err := writer.Flush(); err != nil {
+			t.Fatal(err)
+		}
+
+		if len(buf.String()) != testCase.expected {
+			t.Errorf("Failure in test case %d. Bridge received %d bytes when expecting %d. \n buf: %v", i, len(buf.String()), testCase.bridgeCapacity, buf)
```

Comment:
The docs say: `Errorf is equivalent to Logf followed by Fail.`

I like to use `Errorf` instead of `Fatalf` exclusively because it runs all my test cases every time giving me extra feedback.